### PR TITLE
Upgrade to python3.11

### PR DESCRIPTION
**Pull request by rebkwok**
Also added pyupgrade (it'll probably go away soon with ruff, but it was useful for this), which made a bunch of small changes, some of which date back to older-than-3.8 pythons.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -120,6 +123,10 @@ check *args: devenv
     $BIN/black --check .
     $BIN/isort --check-only --diff .
     $BIN/flake8
+    $BIN/pyupgrade --py311-plus \
+    $(find builder codelists conversions mappings opencodelists services superusers  -name '*.py' -type f) \
+    $(find coding_systems -not -path 'coding_systems/snomedct/parser_utils/*' -name '*.py' -type f)
+
```
Comment (StevenMaude) [inline justfile:129]: suggestion: It might be useful to comment why these files are selected and excluded in this way.

**Review by StevenMaude — APPROVED** [review]

File: justfile [hunk 2]
```diff
@@ -28,6 +28,9 @@ virtualenv:
     # allow users to specify python version in .env
     PYTHON_VERSION=${PYTHON_VERSION:-python3.11}
 
+    # Error if venv does not contain the version of Python we expect
+    test -e $BIN/$PYTHON_VERSION || { echo "Did not find $PYTHON_VERSION in $VIRTUAL_ENV (try deleting and letting it re-build)"; exit 1; }
```
Comment (inglesp) [inline justfile:32]: Try deleting the virtualenv?

File: codelists/tests/test_api_decorators.py [hunk 3]
```diff
@@ -9,7 +9,7 @@ def test_require_authentication_success(client, user, user_codelist):
             "coding_system_database_alias": "snomedct_test_20200101",
         },
         content_type="application/json",
-        HTTP_AUTHORIZATION=f"Token {user.api_token}".encode("utf8"),
+        HTTP_AUTHORIZATION=f"Token {user.api_token}".encode(),
```
Comment (inglesp) [inline codelists/tests/test_api_decorators.py:12]: TIL!

File: coding_systems/snomedct/import_data.py [hunk 4]
```diff
@@ -100,15 +100,13 @@ def build_sql(model):
...
 
-    return """
+    return f"""
     INSERT INTO {table_name}({cols})
       VALUES ({params})
     ON CONFLICT(id) DO UPDATE SET
       {updates}
     WHERE excluded.effective_time > {table_name}.effective_time;
-    """.format(
-        **locals()
```
Comment (inglesp) [inline coding_systems/snomedct/import_data.py:113]: Neat.

**Review by inglesp — APPROVED** [review]
Great, thanks!

**Pull request merged**